examples/neighbors/plot_lof_outlier_detection.py

```diff
@@ -61,13 +72,21 @@
     s=1000 * radius,
     edgecolors="r",
     facecolors="none",
-    label="Outlier scores",
+)
+# Create dummy Line2D objects for legend
+dummy = Line2D(
+    [0],
+    [0],
+    marker="o",
+    markersize=10,
+    markeredgecolor="r",
+    markerfacecolor="none",
+    linestyle="none",
 )
 plt.axis("tight")
 plt.xlim((-5, 5))
 plt.ylim((-5, 5))
 plt.xlabel("prediction errors: %d" % (n_errors))
-legend = plt.legend(loc="upper left")
-legend.legendHandles[0]._sizes = [10]
-legend.legendHandles[1]._sizes = [20]
+plt.legend([scatter, dummy], ["data points", "outlier scores"])
```

Comment:
```suggestion
marker_size = 20
def update_legend_marker_size(handle, orig):
    handle.update_from(orig)
    handle.set_sizes([marker_size])

plt.legend(
    handler_map={scatter: HandlerPathCollection(update_func=update_legend_marker_size)}
)
```

Reply:
This is somehow linked with the legend guide from matplolib: https://matplotlib.org/stable/tutorials/intermediate/legend_guide.html#legend-handlers

At least the code is more explicit on what we try to achieve.